http/src/v0/block.rs

```diff
@@ -102,8 +81,85 @@ pub fn put<T: IpfsTypes>(
     path!("block" / "put")
         .and(with_ipfs(ipfs))
         .and(query::<PutQuery>())
-        .and(multipart::form())
-        .and_then(put_query)
+        .and(warp::header::<Mime>("content-type"))
+        .and(warp::body::stream())
+        .and_then(inner_put)
+}
+
+async fn inner_put<T: IpfsTypes>(
+    ipfs: Ipfs<T>,
+    opts: PutQuery,
+    mime: Mime,
+    body: impl Stream<Item = Result<impl Buf, warp::Error>> + Unpin,
+) -> Result<impl Reply, Rejection> {
+    let boundary = mime.get_param("boundary").map(|v| v.to_string()).unwrap();
+    let mut stream = MultipartStream::new(boundary, body.map_ok(|mut buf| buf.to_bytes()));
+
+    // store the first good field here
+    let mut buffer = Vec::new();
+
+    loop {
+        match stream.try_next().await {
+            Ok(Some(mut field)) => {
+                // [ipfs http api] says we should expect a "data" but examples use "file" as the
+                // form field name. conformance tests also use latter.
```

Comment:
```suggestion
                // form field name. conformance tests also use "data".
```

Reply:
Apparently this had changed just between the two different versions I was testing (by ci here and by locally).